pkg/azure/test_utils.go

```diff
@@ -36,6 +36,8 @@ import (
 )
 
 const (
+	fakeNsgName                              = "test-nsg-name"
+	fakeNsgID                                = "a/b/" + fakeNsgName
```

Comment:
Maybe make this properly formatted like `uriPrefix + "Microsoft.Network/networkSecurityGroups/" + name`? In fact, isn't this redundant with `validSecurityGroupID` below?

Reply:
actually, it is